verifier/detail_template.html

```diff
@@ -150,145 +155,153 @@
+        if (chart && chart_data) {
+            let chart_options = {
+                legend: {position: 'bottom', maxLines: 3},
+                isStacked: true,
+                width: 600, height:200, bar: {groupWidth: '90%' }
+            };
+            chart.draw(chart_data, chart_options);
         }
-        // Do something with the selected set.
-        // Get id of the correct count item.
-        const class_name = box_class;  // get from a parent.
-        const element_name = 'selectedCount';
-        const selected_count_items = document.getElementsByName(element_name);
+    }
 
-        // Spread this out into an array for filtering.
+    // Constants for display of tristate widget
+    const question_mark = '\u2753';
+    const check_mark = '\u2705';
+    const cross_mark = '\u274C';
+
+    function widgetChanged(widget) {
+        // Update the group selected with the intersection or inclusion of label sets
+        const class_name = widget.className;
+        const test_data = test_results[class_name];
+        
+        // Start with all the tests, intersecting or removing data as needed.
+        test_data.selected_set = new Set(test_data.all_labels);
+
+        const selected_count_items = document.getElementsByName('selectedCount');
+        
+        // Update the number of results size
+        let newSize = test_data.selected_set == null ? 0 : test_data.selected_set.size;
         const output = [...selected_count_items].filter(elem => elem.className == class_name);
-        const newSize = test_data.selected_set == null ? 0 : test_data.selected_set.size;
+        if (output) {
+            // Set the current count of items
+            output[0].innerHTML = output[0].innerText = newSize;
+        }
+
+        let excluded_set = new Set();  // Remove these label after the loop.
+        
+        // For each widget, if active, update the state of the selected and exclued sets
+        for (let index in test_data.check_boxes) {
+            const widget = test_data.check_boxes[index];
+            const label_set = test_data.widget_label_sets[index];
+
+            // This is a tristate item.
+            const which_state = widget.value;
+            
+            // Choose the value depending on the status of the control
+            if (which_state == check_mark) {
+                // Update by intersection with this set.
+                test_data.selected_set =
+                    new Set([...test_data.selected_set].filter((x) => label_set.has(x)));
+            } 
+            else if (which_state == cross_mark) {
+                // Update by removing all these labels, i.e., intersect with the inverse
+                for (let label of label_set) {
+                    excluded_set.add(label);
+                }
+            }
+        }
+        
+        // Remove the excluded values from selected set.
+        excluded_set.forEach((label) => {
+            test_data.selected_set.delete(label);
```

Comment:
Comparing this line on 272 to line 260, it seems as if finding the intersection of the checked options is done one way (using `.filter((x) => label_set.has(x))`) that isn't followed when doing the intersection of the negation of the crossed/excluded options (using `.forEach((label) => {test_data.selected_set.delete(label);})`)

So long as it all works correctly, it's fine. And if you're using sets, and your code is always recomputing set intersections whenever the UI is updated after an updated selection of checks and crosses of key-value pairs, then I suppose it should all work.